### PR TITLE
Fix Github response memoization

### PR DESCRIPTION
**Pull request by nateberkopec**
Changing options blows the cache.
+ Add page change method
+ Move last page? to Resource
+ Remove state setting on job, b/c it's just the default.

### CODE REVIEW

File: app/models/github_fetcher/resource.rb [hunk 1]
```diff
@@ -12,18 +12,24 @@ def initialize(options)
 
     # Generally not over-ridden
     def as_json
-      @as_json ||= response.json_body
+      @as_json ||= Hash.new do |h, key|
+                     h[key] = response.json_body
+                   end
+      @as_json[options]
```
Comment (schneems) [inline app/models/github_fetcher/resource.rb]: I think this is going to blow out memory. For kubernetes we will have 176 keys and just as many bodies in the hash (which are not small). Let's not memoize here.

File: test/jobs/populate_issues_job_test.rb [hunk 2]
```diff
@@ -29,11 +29,11 @@ class PopulateIssuesJobTest < ActiveJob::TestCase
 
   test "#populate_multi_issues stores error info when fails" do
     repo = repos(:issue_triage_sandbox)
-    fetcher_double = OpenStruct.new(error?: true, error_message: 'something went wrong')
-
-    repo.stub(:issues_fetcher, -> (_) { fetcher_double }) do
-      PopulateIssuesJob.perform_now(repo)
-      assert_equal repo.github_error_msg, 'something went wrong'
+    def repo.issues_fetcher
+      OpenStruct.new(error?: true, error_message: 'something went wrong', page: 1)
```
Comment (schneems) [inline test/jobs/populate_issues_job_test.rb]: I like this. Who needs mocha 😜

File: app/models/github_fetcher/resource.rb [hunk 3]
```diff
@@ -12,18 +12,24 @@ def initialize(options)
...
-    # Generally not over-riden
+    # Generally not over-ridden
     def response
-      @response ||= begin
-                      GitHubBub.get(api_path, options)
-                    rescue GitHubBub::RequestError => e
-                      @error = e
-                      @error_message = e.message
-                      null_response(e)
+      @response ||= Hash.new do |h, key|
```
Comment (schneems) [inline app/models/github_fetcher/resource.rb]: Same with hash memoization here, will blow out memory

**Comment by nateberkopec** [comment]
.@schneems pointed out the response hash will get huge when paginating a few hundred times, not a good idea. need to blow it away completely.

**Comment by nateberkopec** [comment]
Fixed

**Pull request merged**